File: demo-material3/lib/features/peer_evaluation/domain/models/evaluation.dart
// Pure Dart — NO Flutter imports
class Evaluation {
  final String id;
  final String evaluatorId;
  final String peerId;
  final String assessmentId;
  /// criteriaId → score (2, 3, 4, or 5)
  final Map<String, int> scores;

  const Evaluation({
    required this.id,
    required this.evaluatorId,
    required this.peerId,
    required this.assessmentId,
    required this.scores,
  });
}

File: demo-liquid-glass/DemoLiquidGlass/domain/models/Evaluation.swift
// domain/models/Evaluation.swift
// Pure Swift struct — NO SwiftUI, NO Foundation imports

/// One student's evaluation of one peer, scored per-criteria.
/// Valid scores are 2, 3, 4, or 5 (PeerAssess scale).
public struct Evaluation: Identifiable, Equatable {
    public let id: String
    public let assessmentId: String
    public let evaluatorId: String
    public let peerId: String
    /// key: criteriaId, value: score in {2, 3, 4, 5}
    public let scores: [String: Int]

    public init(
        id: String,
        assessmentId: String,
        evaluatorId: String,
        peerId: String,
        scores: [String: Int]
    ) {
        self.id = id
        self.assessmentId = assessmentId
        self.evaluatorId = evaluatorId
        self.peerId = peerId
        self.scores = scores
    }
}
